doc/installationguide.md

```diff
@@ -21,21 +21,51 @@ These are the parameters that can be configured in the global section:
         port: 4041
     	}
 ```
-* **stats**: configure the periodic collection of statistics. Use `interval` in miliseconds to set the time between stats writings.
+* **stats**: configure the periodic collection of statistics. Use `interval` in milliseconds to set the time between stats writings.
 ```
     stats: {
         interval: 100
     }
 ```
-* **authentication**: authentication data, for use in retrieving tokens for devices with a trust token (just needed in scenarios with security enabled in the Context Broker side). E.g.:
-```
-	{
-        host: 'localhost',
-        port: '5000',
-        user: 'iotagent',
-        password: 'iotagent'
-	}
-```
+* **authentication**: authentication data, for use in retrieving tokens for devices with a trust token (required in scenarios with security enabled in the Context Broker side). 
+  Currently, two authentication provider are supported: `keystone` and `oauth2`. 
+  In `keystone` based authentication, the `trust` associated to the `device` or `deviceGroup` is a token representing a specific user and his rights on a given domain (i.e.
+  combination of `fiware-service` and `service-path`). The authentication process use the trust delegation workflow to check if the trust provided is valid, in which
+  case return a `x-subject-token` that can be used to authenticate the request to the context broker. Required parameters are: the `url` of the keystone to be used
+  (alternatively `host` and `port` but if you use this combination, the IoT Agent will assume that the protocol is HTTP), the `user` and `password` to which it is delegated
+  the `trust` verification.
+  E.g.:
+    ```
+    {
+          enabled: true,
+          url: 'https://localhost:5000',
+          type: 'keystone',
+          user: 'iotagent',
+          password: 'iotagent'
+    }
+    ```
+  In `oauth2` based authentication, the `trust` associated to the `device` or `deviceGroup` is a `refresh_token` issued by a specific user for the context broker client. 
```

Comment:
context broker -> Context Broker

(to use the same style used above)

Reply:
(In general apply the same to other occurrences of the same in the document)

Reply:
fix by 9ea2773